docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+
+### `system-redis`
+This contains Queues for jobs to be processed in background. These are ephemeral and are deleted once processed.
+
+## Backup Procedures
+
+### `system-mysql`
+
+1. Execute MySQL Backup Command
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'export MYSQL_PWD=${MYSQL_ROOT_PASSWORD}; mysqldump --single-transaction -hsystem-mysql -uroot system' | gzip > system-mysql-backup.gz
+   ```
+
+### `system-storage`
+
+1. Archive the system-storage files to another storage.
+    ```
+    oc rsync $(oc get pods -l 'deploymentConfig=system-app' -o json | jq '.items[0].metadata.name' -r):/opt/system/public/system ./local/dir
+    ```
+
+### `zync-database`
+
+1. Execute Postgres Backup Command
+   ```
+    oc rsh $(oc get pods -l 'deploymentConfig=zync-database' -o json | jq '.items[0].metadata.name' -r) bash -c 'pg_dumpall -c --if-exists' | gzip > zync-database-backup.gz
+    ```
+
+### `backend-redis`
+
+1. Backup the dump.rb file from redis
+   ```
+   oc cp $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb ./backend-redis-dump.rdb
+   ```
+
+### `system-redis`
+
+1. Backup the dump.rb file from redis
+
+   ```
+   oc cp $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb ./system-redis-dump.rdb
+   ```
+
+## Restore Procedures
+
+### `system-mysql`
+
+1. Copy the MySQL dump to the system-mysql pod
+   ```
+   oc cp ./system-mysql-backup.gz $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq '.items[0].metadata.name' -r):/var/lib/mysql
+   ```
+2. Decompress the Backup File
+    ```
+    oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'gzip -d ${HOME}/system-mysql-backup.gz'
+    ```
+3. Restore the MySQL DB Backup file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-mysql' -o json | jq -r '.items[0].metadata.name') bash -c 'export MYSQL_PWD=${MYSQL_ROOT_PASSWORD}; mysql -hsystem-mysql -uroot system < ${HOME}/system-mysql-backup'
+   ```
+
+### `system-storage`
+
+1. Restore the Backup file to system-storage
+   ```
+   oc rsync ./local/dir $(oc get pods -l 'deploymentConfig=system-app' -o json | jq '.items[0].metadata.name' -r):/opt/system/public/system
+   ```
+
+### `zync-database`
+
+1. Copy the Zync Database dump to the zync-database pod
+   ```
+   oc cp ./zync-database-backup.gz $(oc get pods -l 'deploymentConfig=zync-database' -o json | jq '.items[0].metadata.name' -r):/var/lib/pgsql/
+   ```
+2. Decompress the Backup File
+    ```
+    oc rsh $(oc get pods -l 'deploymentConfig=zync-database' -o json | jq -r '.items[0].metadata.name') bash -c 'gzip -d ${HOME}/zync-database-backup.gz'
+    ```
+3. Restore the MySQL DB Backup file
```

Comment:
Zync DB is Postgres not MySQL, isn't it? The command described is already OK (psql)

Reply:
Corrected on 7f843576b3e3ceae8c8ef20813ab8b9c89276bb0

Reply:
Cool!